src/vs/workbench/browser/parts/activitybar/activitybarPart.ts

```diff
@@ -462,7 +462,9 @@ export class ActivitybarPart extends Part implements IActivityBarService {
 			const index = this.pinnedViewlets.indexOf(viewletId);
 			this.pinnedViewlets.splice(index, 1);
 
-			this.updateViewletSwitcher();
+			if (this.viewletSwitcherBar) {
```

Comment:
@nicksnyder should we not just move this check into the `updateViewletSwitcher` method? There are many more places where we call it.